src/html.rs

```diff
@@ -135,12 +138,11 @@ impl HtmlMsgParser {
                 }
                 MimeMultipartType::Single => {
                     let mimetype = mail.ctype.mimetype.parse::<Mime>()?;
-                    if mimetype == mime::TEXT_HTML {
+                    if mimetype == mime::TEXT_HTML && self.html.is_empty() {
```

Comment:
I am not sure about collapsing the conditions like that. It's correct because the following `if` will fail, but now we check `mimetype == mime::TEXT_PLAIN` if `self.html.is_empty()` is false, even though we already checked that it's `mime::TEXT_HTML`.

Reply:
i changed that because clippy complained wrt uncollapsed statements, however, having a closer look clippy complains only about the second once. i'll do a commit.